client/signup/config/flows.js

```diff
@@ -16,32 +17,49 @@ import { addQueryArgs } from 'calypso/lib/url';
 import { generateFlows } from 'calypso/signup/config/flows-pure';
 import stepConfig from './steps';
 
-function getCheckoutUrl( dependencies, localeSlug, flowName ) {
+function constructBackUrlFromPath( path ) {
+	if ( config( 'env' ) !== 'production' ) {
+		const protocol = config( 'protocol' ) ?? 'https';
+		const port = config( 'port' ) ? ':' + config( 'port' ) : '';
+		const hostName = config( 'hostname' );
+
+		return `${ protocol }://${ hostName }${ port }${ path }`;
+	}
+
+	return `https://${ config( 'hostname' ) }${ path }`;
+}
+
+function getCheckoutUrl( dependencies, localeSlug, flowName, destination ) {
 	let checkoutURL = `/checkout/${ dependencies.siteSlug }`;
 
 	// Append the locale slug for the userless checkout page.
 	if ( 'no-site' === dependencies.siteSlug && true === dependencies.allowUnauthenticated ) {
 		checkoutURL += `/${ localeSlug }`;
 	}
 
-	let checkoutBackUrl = `https://${ config( 'hostname' ) }/start/${ flowName }/domain-only`;
-	if ( config( 'env' ) !== 'production' ) {
-		const protocol = config( 'protocol' ) ?? 'https';
-		const port = config( 'port' ) ? ':' + config( 'port' ) : '';
-		const hostName = config( 'hostname' );
-
-		checkoutBackUrl = `${ protocol }://${ hostName }${ port }/start/${ flowName }/domain-only`;
+	let isDomainOnly;
+	// TODO:
+	// the domain only flow has special rule. Ideally they should also be configurable in flows-pure.
+	if ( [ 'domain', 'domain-for-gravatar' ].includes( flowName ) ) {
+		isDomainOnly = 1;
+		destination = `/start/${ flowName }/domain-only`;
```

Comment:
Can we pls avoid reassigning/overriding the function parameter this way? 
I'm a little surprised we haven't enforced a rule for this https://eslint.org/docs/latest/rules/no-param-reassign 🤔

Reply:
Great call-out. I've updated it by https://github.com/Automattic/wp-calypso/pull/92168/commits/447bea2f28ac6218e15d0bc0a8792c91ebc4136f, which is similar to your suggestion below. Does it look good to you? :)

Reply:
Yep. Better 🙌